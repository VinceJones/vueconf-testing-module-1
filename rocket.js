export function createRocket () {

  const engines = 4
  const fuelConsumptionPerEngine = 10

  return {
    started: false,
    fuel: 0,
    startEngine () {

      if( ! this.hasSufficientFuel()) {
        throw Error('Insufficient fuel')
      }

      this.started = true
      console.log('ðŸ”¥ - Engine started')
    },

    stopEngine () {
      this.started = false
    },

    takeOff() {
      if( ! this.started ) {
        throw Error('Engine must be started before taking off')
      }

      console.log('ðŸš€ Taking off - Bon Voyage!')
    },

    addFuel (rocketFuel) {
      this.fuel += rocketFuel
    },

    hasSufficientFuel() {
      return this.fuel >= engines * fuelConsumptionPerEngine
    }
  }
}