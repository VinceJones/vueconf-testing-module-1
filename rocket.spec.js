import { createRocket } from './rocket';

let rocket;

beforeEach(() => {
  rocket = createRocket();
})

// 5.2
it('can add fuel', () => {
  rocket.addFuel(100);

  expect(rocket.fuel).toEqual(100);
});

// 5.3
it('must have 40 tonnes of rocket fuel to have sufficient', () => {
  rocket.fuel = 39
  expect(rocket.hasSufficientFuel()).toBeFalsy();

  rocket.fuel = 40;
  expect(rocket.hasSufficientFuel()).toBeTruthy();
});

// 5.4
it('cannot start engine without fuel', () => {
  expect(() => rocket.startEngine()).toThrow('Insufficient fuel');
});

// 5.5
it('can start engine with sufficient fuel', () => {
  rocket.addFuel(200);

  expect(() => rocket.startEngine()).not.toThrow();
});

// 6.2
it('matches snapshot', () => {
  expect(rocket).toMatchSnapshot();
})