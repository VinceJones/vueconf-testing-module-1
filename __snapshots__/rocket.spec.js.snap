// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Object {
  "addFuel": [Function],
  "fuel": 0,
  "hasSufficientFuel": [Function],
  "startEngine": [Function],
  "started": true,
  "stopEngine": [Function],
  "takeOff": [Function],
}
`;
